consumer/pdata/logs.go

```diff
@@ -15,12 +15,106 @@
 package pdata
 
 import (
+	"fmt"
+
 	"go.opentelemetry.io/collector/internal"
 	otlpcollectorlog "go.opentelemetry.io/collector/internal/data/protogen/collector/logs/v1"
 	otlplogs "go.opentelemetry.io/collector/internal/data/protogen/logs/v1"
 )
 
-// This file defines in-memory data structures to represent logs.
+// LogsDecoder is an interface to decode bytes into protocol-specific data model.
+type LogsDecoder interface {
```

Comment:
Can we put these marshalling/decoding/etc interfaces in a separate package or separate file?

Reply:
We can discuss about separate files, but more importantly is do we want same package or separate? Advantage for same package is that if we also put OTLP favors in pdata we don't need the InternalRep at all

Reply:
@tigrannajaryan Are you OK with having them in the same package for the reason above?

Reply:
If you are OK, I would merge the PR as is, and will followup since is a non-breaking change, just to ensure we release these interfaces in the next release so I can start using them in contrib.